spec/services/case_extend_sar_deadline_service_spec.rb

```diff
@@ -0,0 +1,172 @@
+
+          expect(result).to eq :ok
+          expect(sar_case.external_deadline).to eq deadline + max_extension.days
+          expect(sar_case.external_deadline).to be < DateTime.now
+        end
+      end
+    end
+
+    context 'validate' do
+      context 'extension period' do
+        context 'is missing' do
+          subject! { new_service(manager, sar_case, nil).call }
+
+          it { is_expected.to eq :validation_error }
+          it {
+            expect(sar_case.errors[:extension_period])
+              .to eq ["can't be blank"]
+          }
+        end
+
+        context 'is past statutory SAR extension limit' do
+          let!(:extension_period) { max_extension + 5 }
+          subject! {
+            new_service(
+              manager,
+              sar_case,
+              extension_period
+            ).call
+          }
+
+          it { is_expected.to eq :validation_error }
+          it {
+            expect(sar_case.errors[:extension_period])
+              .to eq ["can't be more than 60 days beyond the initial deadline"]
+          }
+        end
+
+        context 'is before the final deadline' do
+          subject! { new_service(manager, sar_case, -1).call }
+
+          it { is_expected.to eq :validation_error }
+          it {
+            expect(sar_case.errors[:extension_period])
+              .to eq ["can't be before the final deadline"]
+          }
+        end
+
+        context 'is within limit' do
+          subject! { new_service(manager, sar_case, max_extension).call }
+
+          it { is_expected.to eq :ok }
+        end
+      end
+
+      context 'reason' do
+        subject! { new_service(manager, sar_case, 1, reason: '').call }
+
+        it { is_expected.to eq :validation_error }
+        it {
+          expect(sar_case.errors[:reason_for_extending])
+            .to eq ["can't be blank"]
+        }
+      end
+
+      context 'exception' do
+        let(:service) { new_service(manager, sar_case, 3) }
+
+        it 'rolls-back changes' do
+          allow(sar_case).to receive(:update!).and_throw(RuntimeError)
+
+          service.call
+          transitions = sar_case.transitions.where(
+            event: 'extend_sar_deadline'
+          )
+
+          expect(transitions.any?).to be false
+        end
+
+        it 'sets result to :error and returns :error' do
+          allow(sar_case).to receive(:update!).and_throw(RuntimeError)
+
+          result = service.call
+
+          expect(result).to eq :error
+          expect(service.result).to eq :error
+        end
+      end
+    end
+  end
+
+  context '#new_extension_deadline' do
+    context 'calculates extended date with calendar days' do
+      it 'uses the original date before first extension' do
+        service = new_service(manager, sar_case, 3)
+
+        expect(service.send(:new_extension_deadline, 3))
+          .to eq(initial_deadline + 3.days)
+
+        service.call # required to create new deadline
```

Comment:
What do you mean by this comment? Also not sure of the setup up here, why are we sending the methods to the service shouldn't we create the service with the variables?

Reply:
I'm using `service.send(:new_extension_deadline, 3)` because `new_extension_deadline` is a private method.

Reply:
but isn't it called from the initialise? shouldn't we create the service with the variables?